pkg/repository/backend_postgres.go

```diff
@@ -72,8 +74,11 @@ func NewBackendPostgresRepository(config types.PostgresConfig, eventRepo EventRe
 		eventRepo: eventRepo,
 	}
 
-	if err := repo.migrate(); err != nil {
-		log.Fatalf("failed to run backend migrations: %v", err)
+	if release, err := obtainMigrationLock("postgres"); err == nil {
+		defer release()
+		if err := repo.migrate(); err != nil {
+			log.Fatalf("Failed to run backend migrations: %v", err)
+		}
```

Comment:
Ignoring the error here when its not nil means we are skipping for now if we fail to acquire the lock?

Reply:
If we were to err here, the startup would fail entirely, which seems like that might keep us in a more consistent state. The new logic wouldn't potentially be out ahead of the migration being complete. Also, it should retry rolling out the update anyway right?

Reply:
Another benefit is that we could do the locking logic in `NewGateway` instead of passing it down into repo.

Reply:
Hm yeah, I'll move it out and keep it in `NewGateway`.

Can you explain more what you mean in your 2nd comment? I'm not sure I follow.

Reply:
Made migrate a public function and now its being wrapped by the lock.